src/kudu/consensus/leader_election.cc

```diff
@@ -70,16 +84,16 @@ VoteCounter::VoteCounter(int num_voters, int majority_size)
   CHECK_GT(majority_size_, 0);
 }
 
-Status VoteCounter::RegisterVote(const std::string& voter_uuid, ElectionVote vote,
-                                 bool* is_duplicate) {
+Status VoteCounter::RegisterVote(const std::string& voter_uuid,
+                                 VoteInfo vote_info, bool* is_duplicate) {
   // Handle repeated votes.
   if (PREDICT_FALSE(ContainsKey(votes_, voter_uuid))) {
     // Detect changed votes.
-    ElectionVote prior_vote = votes_[voter_uuid];
-    if (PREDICT_FALSE(prior_vote != vote)) {
+    VoteInfo prior_vote_info = votes_[voter_uuid];
```

Comment:
Prefer using find() method of std::map which is safer. [] can throw or default construct on non-existing key.
```suggestion
    auto prior_vote_info = votes_.find(voter_uuid);
```

Reply:
We are checking presence of the key before accessing so I thought this was okay. Will change it to use `at`. Should be much cleaner.